### PR TITLE
Training & Inference code for FAcodec

### PR DESCRIPTION
**Pull request by Plachtaa**
## ✨ Description

Training & Inference code for FAcodec, include link to pretrained checkpoint

## 🚧 Related Issues

None

## 👨‍💻 Changes Proposed

- [x] Added new requirements to `env.sh`
- [x] Added `codec` directory for `bins`, `models`, `egs`

## 🧑‍🤝‍🧑 Who Can Review?

[@zhizhengwu]

## 🛠 TODO

- [ ] To be added...

## ✅ Checklist

- [ ]  Code has been reviewed
- [ ]  Code complies with the project's code standards and best practices
- [ ]  Code has passed all tests
- [ ]  Code does not affect the normal use of existing features
- [ ]  Code has been commented properly
- [ ]  Documentation has been updated (if applicable)
- [ ]  Demo/checkpoint has been attached (if applicable)

### CODE REVIEW

File: models/codec/facodec/modules/attentions.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
+import copy
```
Comment (RMSnow) [inline models/codec/facodec/modules/attentions.py:1]: Add the copyright information

File: models/codec/facodec/modules/layers.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
+import math
```
Comment (RMSnow) [inline models/codec/facodec/modules/layers.py:1]: Add the copyright information

File: modules/dac/model/base.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
+import math
```
Comment (RMSnow) [inline modules/dac/model/base.py:1]: Add the copyright information

File: modules/dac/model/dac.py [hunk 4]
```diff
@@ -0,0 +1,389 @@
+import math
```
Comment (RMSnow) [inline modules/dac/model/dac.py:1]: Add the copyright information

File: modules/dac/model/discriminator.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
+import torch
```
Comment (RMSnow) [inline modules/dac/model/discriminator.py:1]: Add the copyright information

File: modules/dac/model/encodec.py [hunk 6]
```diff
@@ -0,0 +1,292 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (RMSnow) [inline modules/dac/model/encodec.py:1]: Hi @jiaqili3, please help to revise the copyright information for such external modules

File: modules/dac/nn/__init__.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+from . import layers
```
Comment (RMSnow) [inline modules/dac/nn/__init__.py:1]: Add the copyright information

File: modules/dac/nn/layers.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import numpy as np
```
Comment (RMSnow) [inline modules/dac/nn/layers.py:1]: Add the copyright information

File: modules/dac/nn/loss.py [hunk 9]
```diff
@@ -0,0 +1,381 @@
+import typing
```
Comment (RMSnow) [inline modules/dac/nn/loss.py:1]: Add the copyright information

File: modules/dac/nn/quantize.py [hunk 10]
```diff
@@ -0,0 +1,262 @@
+from typing import Union
```
Comment (RMSnow) [inline modules/dac/nn/quantize.py:1]: Add the copyright information

File: egs/codec/FAcodec/README.md
Comment (RMSnow) [inline egs/codec/FAcodec/README.md:1]: Hi @HeCheng0625, please help to polish this readme. Can we release FACodec's checkpoint in Amphion?

**Review by RMSnow — CHANGES_REQUESTED** [review]
Thanks for the great works and high-quality code from @Plachtaa! There are some files need to be added the copyright information. Let @HeCheng0625 double check the readme of FACodec!

**Comment by Plachtaa** [comment]
Hi @RMSnow thanks a lot for reviewing the code! I apologize for the late reply. I have just added copyright statements to all newly added files and source of code where necessary. Please help me check whether the format is correct.
Thanks!

**Review by jiaqili3 — APPROVED** [review]
Looks good to me.

File: egs/codec/FAcodec/train.sh
Comment (xinkez) [inline egs/codec/FAcodec/train.sh:1]: No "exp_libritts.json" file?

Comment (Plachtaa) [reply egs/codec/FAcodec/train.sh:1]: For ease of data preparation, dataset is denoted as a single directory, where audio samples are recursively searched and loaded, as no transcriptions are required.

**Comment by jiaqili3** [comment]
Hi @Plachtaa , could you use black formatting to format the code to pass the code style check? The command is just "black .", thanks!

**Comment by Plachtaa** [comment]
> Hi @Plachtaa , could you use black formatting to format the code to pass the code style check? The command is just "black .", thanks!

Sorry for late reply, I have reformatted all changed & new files with black formatter
